File: iosApp/iosApp/ProfileSetup/ProfileSetupAllSetStep.swift
import SwiftUI

/// The completion screen - no category badge, no form fields. "ProfileSetupAllSetIllustration"
/// is cropped directly from the reference design (a shield + florals, not an SF Symbol
/// like the rest of this screen's icons - matches the reference's original raster asset
/// instead of approximating it, which read as noticeably flatter than the source).
struct ProfileSetupAllSetStep: View {
    var body: some View {
        VStack {
            Text(String(localized: "profile_setup_all_set_title"))
                .font(CyraFont.headlineMedium())
                .foregroundColor(.cyraOnSurface)
                .multilineTextAlignment(.center)
            Spacer().frame(height: 8)
            Text(String(localized: "profile_setup_all_set_subtitle"))
                .font(CyraFont.bodyMedium())
                .foregroundColor(.cyraOnSurfaceVariant)
                .multilineTextAlignment(.center)
            Spacer().frame(height: 24)
            Image("ProfileSetupAllSetIllustration")
                .resizable()
                .scaledToFit()
                .frame(maxWidth: .infinity)
            Spacer().frame(height: 24)
            HStack(spacing: 12) {
                Image(systemName: "lock")
                    .foregroundColor(.cyraOnSurfaceVariant)
                VStack(alignment: .leading, spacing: 2) {
                    Text(String(localized: "profile_setup_all_set_privacy_title"))
                        .font(CyraFont.titleSmall())
                        .foregroundColor(.cyraOnSurface)
                    Text(String(localized: "profile_setup_all_set_privacy_subtitle"))
                        .font(CyraFont.bodySmall())
                        .foregroundColor(.cyraOnSurfaceVariant)
                }
                Spacer()
            }
            .padding(16)
            .background(RoundedRectangle(cornerRadius: 16).fill(Color.cyraPrimaryContainer.opacity(0.4)))
        }
    }
}

#Preview {
    ProfileSetupAllSetStep()
        .padding(24)
        .cyraThemed()
}

File: androidApp/src/main/kotlin/subha/app/cyra/ui/profilesetup/ProfileSetupAllSetStep.kt
package subha.app.cyra.ui.profilesetup

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import subha.app.cyra.R
import subha.app.cyra.ui.components.LockIcon

/**
 * The completion screen - no category badge, no form fields. [R.drawable
 * .profile_setup_all_set_illustration] is cropped directly from the reference design
 * (a shield + florals, not a hand-drawn Canvas icon like the rest of the app's iconography
 * - matches the reference's original raster asset instead of approximating it as a flat
 * vector, which read as noticeably flatter than the source).
 */
@Composable
fun ProfileSetupAllSetStep(modifier: Modifier = Modifier) {
    Column(modifier = modifier, horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = stringResource(R.string.profile_setup_all_set_title),
            style = MaterialTheme.typography.headlineMedium,
            color = MaterialTheme.colorScheme.onSurface,
            textAlign = TextAlign.Center,
        )
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            text = stringResource(R.string.profile_setup_all_set_subtitle),
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
        )
        Spacer(modifier = Modifier.height(24.dp))
        Image(
            painter = painterResource(R.drawable.profile_setup_all_set_illustration),
            contentDescription = null,
            modifier = Modifier.fillMaxWidth(),
        )
        Spacer(modifier = Modifier.height(24.dp))
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(color = MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.4f), shape = RoundedCornerShape(16.dp))
                .padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            LockIcon(modifier = Modifier)
            Spacer(modifier = Modifier.width(12.dp))
            Column {
                Text(
                    text = stringResource(R.string.profile_setup_all_set_privacy_title),
                    style = MaterialTheme.typography.titleSmall,
                    color = MaterialTheme.colorScheme.onSurface,
                )
                Text(
                    text = stringResource(R.string.profile_setup_all_set_privacy_subtitle),
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
        }
    }
}
